capstone/src/main/webapp/profile-loader.js

```diff
@@ -27,25 +27,28 @@ function getStudentInfo() {
       document.getElementsByClassName('profile-pic')[0].src = 'images/profile.jpeg';
     }
   });
+  getInterestedClubList();
 }
 
-/** Fill in inbox template with each club's announcements */
-function addAnnoucements(announcements) {
-  const template = document.querySelector('#inbox-list');
-  for(const announcement of announcements){
-    template.content.querySelector('li').innerHTML = announcement;
-    var clone = document.importNode(template.content, true);
-    document.getElementById('inbox').appendChild(clone);
-  }
+/** Fetch interested club list and display on profile page */
+function getInterestedClubList() {
+  fetch('/interested-clubs').then(response => response.json()).then((interestedClubs) => {
+    displayElements(interestedClubs, '#interested-list', 'interested-club-content');
+  });
 }
 
-/** Fill in club list template with all club names and leave buttons */
-function getClubElements(clubs) {
-  const template = document.querySelector('#club-list');
-  for(const club of clubs){
-    template.content.querySelector('li').innerHTML = getClubContent(club);
+/** Fill list templates with necessary information */
+function displayElements(items, templateId, documentId) {
+  const template = document.querySelector(templateId);
+  for(const item of items){
+    if (templateId == '#club-list') {
+      template.content.querySelector('li').innerHTML = getClubContent(item);
+    }
+    else {
```

Comment:
super nit: single line `} else {`

https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style
"For example, there is no line break after the brace if it is followed by else or a comma."

Reply:
Got it, thanks for pointing these out! :)